src/Glpi/Marketplace/Controller.php

```diff
@@ -288,9 +289,9 @@ public function proxifyPluginArchive(): void
         $dest     = GLPI_TMP_DIR . '/' . mt_rand() . '.' . $filename;
 
         if (!$api->downloadArchive($url, $dest, $this->plugin_key, false)) {
-            http_response_code(500);
-            echo(__('Unable to download plugin archive.'));
-            return;
+            $exception = new HttpException(502);
```

Comment:
Not sure about HTTP 502 here. The server is not completely a "gateway" at this moment, but moreover, this might be a problem if the prod server is unavailable or if there is a config issue specifically for this request. HTTP 500 is fine to me because we know it's application-wide and not infrastructure-wide (which HTTP 502 would be generally used for)

Reply:
I put back the 500 code.